### PR TITLE
yuzu, yuzu-cmd: intergrate feral gamemode support on linux

### LINKED ISSUE #10078: [Feature Request] Add integrated Feral Interactive GameMode support for Linux
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What feature are you suggesting?

This is a feature request for an integrated performance toggle to enable the use of [Feral Interactive GameMode](https://github.com/FeralInteractive/gamemode) on Linux devices.

### Why would this feature be useful?

Currently, to use gamemode users have to manually launch yuzu with `gamemoderun yuzu` or edit their desktop entries to have it automatically run. Having an integrated toggle would allow users to more easily enable the feature and would allow gamemode to be disabled when a game isn't currently running.

### PR DESCRIPTION
**Pull request by xcfrg**
This pull request adds [Feral Gamemode](https://github.com/FeralInteractive/gamemode) support to yuzu on linux, allowing for optimizations to the cpu governor, i/o and process priority, and if using a `-ck` or `-pf` kernel, the kernel scheduler.

Closes #10078

### CODE REVIEW

File: src/yuzu/uisettings.h [hunk 1]
```diff
@@ -85,6 +85,9 @@ struct Values {
     // Discord RPC
     Settings::Setting<bool> enable_discord_presence{true, "enable_discord_presence"};
 
+    // Gamemode
+    Settings::Setting<bool> enable_gamemode{false, "enable_gamemode"};
```
Comment (FearlessTobi) [inline src/yuzu/uisettings.h:89]: Does this need to be a setting? Are there any cases where we don't want this to be active?

Comment (xcfrg) [reply src/yuzu/uisettings.h:89]: Reintroduced

**Comment by v1993** [comment]
Some thoughts on this:

* IMO it would be preferable to use the term "GameMode" rather than "Feral Gamemode" - that's what official documentation uses and I've seen used everywhere else (my initial reaction was "feral gamemode? is that a fork?").
* It might be a good idea to add gamemode repository as a submodule instead of copying file here, but I'm not too sure since only a single header from it is used. Maybe others have stronger opinions on this.
* I'd demote a log on gamemode failure from error to warning at most since it's not even remotely fatal and is actually expected to happen for those who don't use gamemode. You probably can try calling `internal_load_libgamemode` manually to check if gamemode library is present and hide setting in UI as well as don't do enabling/disabling calls if it's not, but this isn't an official part of library's API...
* This likely should also be incorporated into yuzu-cmd, which shouldn't be hard.
* How about checking the return value of `gamemode_request_end`?
* Probably a good idea to handle pausing/unpausing in addition to starting/stopping.

**Comment by xcfrg** [comment]
> Some thoughts on this:
> 
>     * IMO it would be preferable to use the term "GameMode" rather than "Feral Gamemode" - that's what official documentation uses and I've seen used everywhere else (my initial reaction was "feral gamemode? is that a fork?").
> 
>     * It might be a good idea to add gamemode repository as a submodule instead of copying file here, but I'm not too sure since only a single header from it is used. Maybe others have stronger opinions on this.
> 
>     * I'd demote a log on gamemode failure from error to warning at most since it's not even remotely fatal and is actually expected to happen for those who don't use gamemode. You probably can try calling `internal_load_libgamemode` manually to check if gamemode library is present and hide setting in UI as well as don't do enabling/disabling calls if it's not, but this isn't an official part of library's API...
> 
>     * This likely should also be incorporated into yuzu-cmd, which shouldn't be hard.
> 
>     * How about checking the return value of `gamemode_request_end`?
> 
>     * Probably a good idea to handle pausing/unpausing in addition to starting/stopping.

- I have not added the submodule as it is just one header, waiting for other opinions on this
- We can avoid calling the the internal function by using `gamemode_query_status` and checking for a return value of -1, but I am looking for guidance on where I should add this check.
- Everything else has been implemented

**Comment by HurricanePootis** [comment]
Any progress on this?

**Comment by FearlessTobi** [comment]
needs to be rebased.

**Comment by liamwhite** [comment]
Superseded by #11946

**Pull request closed without merging**